### DISCUSSION TITLE
Upgrade Netbox 2.5.8 to latest version integrated with Icinga2

### DISCUSSION THREAD

**Discussion by smallboy69** [netbox-community/netbox#15344] [Q&A]
hi community

hope this email finds you well

please note

i am about to upgrade Netbox to the latest version but i am concerned as i have icinga2 integrated with it...

if anyone has had the experience of upgrading with icinga2

a) will do a backup but since its the first time, i defer to the community experience and insight with such a upgrade...something that i may have missed...

Appreciate your assistance and insight

Thank you

**Comment by candlerb** [comment]
Depends what you mean by "icinga2 integrated". Can you explain further?

There are certainly lots of changes between 2.5.8 and 3.7.3 but how these will affect you depends on what integration points you have been using and how you are using them.

Backing up before each upgrade is fine, but given your Netbox is so ancient I would strongly recommended that you not to do an in-place upgrade at all, but restore your Netbox SQL database to a completely fresh VM and upgrade that.  That's because so many dependencies have also changed (python, postgres, redis, systemd etc) and you're likely to get into knots if you don't upgrade all of those pieces.

You will in any case need to do the upgrade in two stages: first to Netbox v2.11.12, and then to v3.7.3 (latest). So:
* Export the SQL database from 2.5.8
* Build a brand new v2.11.12 VM, and test it (you should use an OS with python 3.8; later versions are not guaranteed to work)
* Restore the SQL database (and media files) into this VM, then re-run `./upgrade.sh`
* Test it
* EIther do an in-place upgrade to v3.7.3; or repeat the whole process:
    * export the SQL database
    * build a brand new v3.7.3 VM and test (this can use python 3.10/3.11; 3.8 currently still works, but support will be dropped soon)
    * import the SQL database
    * run `./upgrade.sh`

At this point you have v3.7.3 running in a completely new VM, so you can test and fix your icinga integration, whilst your production instance is still running.  Of course, any data changes in your production instance after the migration will be lost.

For more info see instructions on [replicating Netbox](https://docs.netbox.dev/en/stable/administration/replicating-netbox/). I omitted the replication of media/images, scripts, reports.
